packages/core/src/util/token.ts

```diff
@@ -71,10 +76,16 @@ export async function getWebidFromTokenPayload(
     // as specified by the Identity Provider.
     // eslint-disable-next-line no-new
     new URL(payload.sub);
-    return payload.sub;
   } catch (e) {
     throw new Error(
       `The token has no 'webid' claim, and its 'sub' claim of [${payload.sub}] is invalid as a URL - error [${e}].`,
     );
   }
+  // The 'sub' claim is a well-formed IRI, hence a valid webId
+  if (typeof payload.azp === "string") {
+    return { webId: payload.sub, clientId: payload.azp };
+  }
+  throw new Error(
+    `The token ${JSON.stringify(payload)} is invalid: it has no 'azp' claim.`,
+  );
```

Comment:
Based on my previous comment, this whole block could be removed and replaced with 
`return { webId, clientId };`. I think type inference should be correct.